builder.go

```diff
@@ -175,25 +192,54 @@ func containsEntry(metRequires []string, entry BuildPlanEntry) bool {
 	return false
 }
 
-type processMap map[string]launch.Process
+// orderedProcesses is a mapping from process types to Processes, it will preserve ordering.
+// processList is the ordered list
+// wwe keep typeToPtr map in order to delete elements when others are overriding them
```

Comment:
```suggestion
// we keep typeToProcess map in order to delete elements when others are overriding them
```

Reply:
Following on https://github.com/buildpacks/lifecycle/pull/520/files#r568185952, this seems to be adding a bit of complexity that is maybe not necessary? Or am I missing something?

Reply:
Ohh I see - it is needed so that we can do `launchMD.FindLastDefaultProcessType()`...

Reply:
Exactly. We need this struct because we care about the order of the processes.